### PR TITLE
Support 'tanzu cluster kubeconfig get' with a vSphere Supervisor

### LINKED ISSUE #2670: Support 'tanzu cluster kubeconfig get' with vSphere Supervisor
**Describe the feature request**

The existing 'cluster' plugin fails fast when it detects a vSphere Supervisor cluster. It would be nice to be able to get both the admin or the pinniped kubeconfig for a workload cluster in a vSphere Supervisor.  

**Describe alternatives you've considered**

A separate CLI is always an option, but that becomes gnarly/fragmented.

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [X] CLI
- [ ] Docs
- [X] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [X] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by mayankbh**
### What this PR does / why we need it

Allow fetching the workload cluster kubeconfig in vSphere with Tanzu 
when a vSphere Supervisor is configured with Pinniped. 

The existing 'cluster kubeconfig get' code path seems to rely on
fetching the admin kubeconfig from the tkg-system namespace on the
management cluster.

Since at this point the user is already authenticated to the management
cluster (as we have a regional cluster client) we can avoid fetching the
management cluster kubeconfig by directly using the user's identity and
fetching the kube-public/pinniped-info ConfigMap.

Misc: Also add some debug logging and plumb the log level configuration
into the aunum/log global logger. I'm not super familiar with logging
conventions in this repo so I'm happy to defer to review judgement.

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/2670

### Which issue(s) this PR fixes
 
N/A

### Describe testing done for PR

Testing:

- Updated unit tests to use fake pinniped-info ConfigMaps in the fake
  client, tests still pass.
- Removed the relevant unit tests that looked for the management
cluster kubeconfig.

### Release note

```release-note

'tanzu cluster kubeconfig get' now works with a vSphere Supervisor.

```

N/A

### PR Checklist

- [X] Squash the commits into one or a small number of logical commits
      
- [X] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [X] Ensure PR contains terms all contributors can understand and links all contributors can access

### Additional information

N/A

#### Special notes for your reviewer

One LoC has been commented out due to a dependency on https://github.com/vmware-tanzu/tanzu-framework/pull/2539

### CODE REVIEW

File: cmd/cli/plugin/cluster/main.go
Comment (ankeesler) [reply cmd/cli/plugin/cluster/main.go:70]: might need to get someone like @anujc25 to take a look

**Review by ankeesler — COMMENTED** [review]
generally looks good. feel free to ping me after you rebase with 2539 for a final review

File: cmd/cli/plugin/cluster/main.go
Comment (mayankbh) [reply cmd/cli/plugin/cluster/main.go:70]: Might relate to https://github.com/vmware-tanzu/tanzu-framework/issues/337

Comment (mayankbh) [reply cmd/cli/plugin/cluster/main.go]: I decided to ditch it for now, might be okay to fix in a followup, the Debugf() will exist but get silently discarded.

**Comment by mayankbh** [comment]
cc @ankeesler just rebased on top of 2539. Working on verifying again but it compiles so I'm optimistic.

**Review by ankeesler — APPROVED** [review]
lgtm from the iam end

**Comment by mayankbh** [comment]
@vuil  @vijaykatam  I think this might need one of your approvals, based on the CODEOWNERS?

(Andrew from the IAM side has approved it fwiw)

**Review by vuil — APPROVED** [review]
LGTM,  thanks for the changes!

**Pull request merged**
